src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -37,57 +39,404 @@ const ActorInit En_Cow_InitVars = {
+            if (this->flags & EN_COW_FLAG_WONT_GIVE_MILK) {
+                this->flags &= ~EN_COW_FLAG_WONT_GIVE_MILK;
+                D_801BDAA4 = 0;
+            } else if ((this->actor.xzDistToPlayer < 150.0f) &&
+                       ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) < 25000) {
+                D_801BDAA4 = 0;
+                this->actionFunc = EnCow_Talk;
+                this->actor.flags |= 0x10000;
+                func_800B8614(&this->actor, globalCtx, 170.0f);
+                this->actor.textId = 0x32C8;
+
+                EnCow_UpdateAnimation(this, globalCtx);
+                return;
+            } else {
+                this->flags |= EN_COW_FLAG_WONT_GIVE_MILK;
+            }
+        } else {
+            this->flags &= ~EN_COW_FLAG_WONT_GIVE_MILK;
+        }
+    }
+
+    if (this->actor.xzDistToPlayer < 150.0f &&
+        ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) < 25000) {
+        if (func_801A5100() == 4) {
+            if (!(gSaveContext.weekEventReg[87] & 1)) {
+                gSaveContext.weekEventReg[87] |= 1;
+                if (func_80114E90() != 0) {
+                    this->actor.textId = 0x32C9;
+                } else {
+                    this->actor.textId = 0x32CA;
+                }
+                this->actor.flags |= 0x10000;
+                func_800B8614(&this->actor, globalCtx, 170.0f);
+                this->actionFunc = EnCow_Talk;
+            }
+        } else {
+            gSaveContext.weekEventReg[87] &= (u8)~1;
+        }
+    }
+
+    EnCow_UpdateAnimation(this, globalCtx);
+}
+
+void EnCow_DoTail(EnCow* this, GlobalContext* globalCtx) {
+    if (this->animationTimer > 0) {
+        this->animationTimer--;
+    } else {
+        this->animationTimer = Rand_ZeroFloat(200.0f) + 40.0f;
+        Animation_Change(&this->skelAnime, &gCowTailIdleAnim, 1.0f, this->skelAnime.curFrame,
+                         Animation_GetLastFrame(&gCowTailIdleAnim), ANIMMODE_ONCE, 1.0f);
+    }
+
+    if (this->actor.xzDistToPlayer < 150.0f &&
+        ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) > 25000) {
+        if (!(this->flags & EN_COW_FLAG_PLAYER_HAS_APPROACHED)) {
+            this->flags |= EN_COW_FLAG_PLAYER_HAS_APPROACHED;
+            if (this->skelAnime.animation == &gCowTailIdleAnim) {
+                this->animationTimer = 0;
+            }
+        }
+    }
+}
 
-extern ColliderCylinderInit D_8099D610;
+void EnCow_Update(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    EnCow* this = THIS;
+    s16 targetX;
+    s16 targetY;
+    Player* player = GET_PLAYER(globalCtx);
 
-extern UNK_TYPE D_060001CC;
-extern UNK_TYPE D_06004010;
-extern UNK_TYPE D_06004348;
+    CollisionCheck_SetOC(globalCtx, &globalCtx2->colChkCtx, &this->colliders[0].base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx2->colChkCtx, &this->colliders[1].base);
```

Comment:
Usually we put globalCtx2 as the function arg to make it easier to swap in and out.

Is it required to use the temps like this? If so, that's very interesting, and one we should bear in mind. @AngheloAlf will want to make a note of this for when the time comes.

Reply:
Yes, this is one of the rare cases where we need `globalCtx2` (or, at least, I couldn't get it to match otherwise).

Reply:
To follow up on this, using the temp isn't actually needed. Assigning `GlobalContext* globalCtx = globalCtx2;` then using `globalCtx` for everything works too.